scubagoggles/Testing/Functional/SmokeTests/smoke_test.py

```diff
@@ -0,0 +1,82 @@
+"""
+smoke_test.py declares a SmokeTest class for ScubaGoggles automation testing.
+"""
+
+import subprocess
+
+import pytest
+
+from smoke_test_utils import (get_output_path,
+                              get_required_entries,
+                              run_selenium,
+                              sample_report_dir,
+                              top_report_url,
+                              verify_all_outputs_exist,
+                              verify_output_type,
+                              verify_scubaresults)
+
+
+class SmokeTest:
+    """
+    Pytest class to encapsulate the following test cases:
+
+    - Generate the correct output files (BaselineReports.html,
+      ScubaResults.json, etc)
+    - Check the content of html files, verify href attributes are
+      correct, etc
+    - Check if ScubaResults.json contains errors in the summary. If errors
+      exist, then either API calls or functions produced exceptions which
+      need to be handled
+    """
+
+    @staticmethod
+    def test_scubagoggles_output(subjectemail: str = None):
+        """
+        Test if the `scubagoggles gws` command generates correct output for
+        all baselines.
+
+        Args:
+            subjectemail: (optional) The email address of a user for the
+            service account.  If not specified, the user's OAuth credentials
+            are used instead.
+        """
+
+        svc_account_option = (f' --subjectemail {subjectemail}' if subjectemail
+                              else '')
+        command = f'scubagoggles gws{svc_account_option} --quiet'
```

Comment:
The subjectemail is not optional for the smoke test, and for some reason including the `= None` made it so that subjectemail would always be None, even if you specified one to pytest. I tested locally, the following change fixed the smoke test for me (tested on Windows).

```suggestion
    def test_scubagoggles_output(subjectemail: str):
        """
        Test if the `scubagoggles gws` command generates correct output for
        all baselines.

        Args:
            subjectemail: The email address of a user for the
            service account.
        """

        command = f'scubagoggles gws --subjectemail {subjectemail} --quiet'
```

Reply:
Because we don’t all have access to a service account, this smoke test must be able to be run with either the service account or the OAuth credentials.  I thought the default value would work correctly (but I couldn’t test it because I don’t have access to a service account).  I’m going to try a different tack with this change and hopefully it’ll work with the service account.